flink/v1.12/flink/src/main/java/com/netease/arctic/flink/write/AutomaticLogWriter.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.netease.arctic.flink.write;
+
+import com.netease.arctic.flink.shuffle.ShuffleHelper;
+import com.netease.arctic.flink.table.ArcticTableLoader;
+import com.netease.arctic.flink.write.hidden.HiddenLogWriter;
+import com.netease.arctic.flink.write.hidden.LogMsgFactory;
+import com.netease.arctic.log.LogData;
+import org.apache.flink.runtime.state.StateInitializationContext;
+import org.apache.flink.runtime.state.StateSnapshotContext;
+import org.apache.flink.streaming.api.graph.StreamConfig;
+import org.apache.flink.streaming.api.operators.Output;
+import org.apache.flink.streaming.api.watermark.Watermark;
+import org.apache.flink.streaming.runtime.streamrecord.StreamRecord;
+import org.apache.flink.streaming.runtime.tasks.StreamTask;
+import org.apache.flink.table.data.RowData;
+import org.apache.iceberg.Schema;
+
+import java.time.Duration;
+import java.util.Properties;
+
+public class AutomaticLogWriter extends ArcticLogWriter {
```

Comment:
Could you add some annotations to describe the overall scenario for this automatic writer? Such as only writing the latest data(in the gap of AUTO_EMIT_LOGSTORE_WATERMARK_GAP from now) into log-store.